src/main/java/uk/gov/justice/hmiprobation/casesampler/controllers/ControllerAdvice.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.hmiprobation.casesampler.controllers
+
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.web.bind.MissingServletRequestParameterException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestClientResponseException
+import uk.gov.justice.hmiprobation.casesampler.dto.ErrorResponse
+import javax.validation.ValidationException
+
+@RestControllerAdvice(basePackages = ["uk.gov.justice.hmipprobation.casesampler.controllers"])
+class ControllerAdvice {
+    @ExceptionHandler(RestClientResponseException::class)
+    fun handleRestClientResponseException(e: RestClientResponseException): ResponseEntity<ByteArray> {
```

Comment:
One to watch out for - RestClientException and its subclasses are only thrown by RestTemplate, not by WebClient should you go that way.